pkg/filter/filter.go

```diff
@@ -108,24 +119,40 @@ func JoinResourceTypeMatchers(ms ...ResourceTypeMatcher) ResourceTypeMatcher {
 
 // ResourceRequirement allows specifying a resource requirement by type and/or name
 type ResourceRequirement struct {
+	// Provides the Name of the resource object
 	v1.LocalObjectReference `json:",inline,omitempty"`
+	// Provides the Group, Version, and Resource values (GVR)
 	ResourceTypeRequirement `json:",inline,omitempty"`
+	// Specifies a set of label requirements to be used as filters for matches
+	metav1.LabelSelector `json:",inline,omitempty"`
 }
 
 // Matches returns true if the specified resource name/GVR matches the requirement
-func (r ResourceRequirement) Matches(name string, gvr schema.GroupVersionResource) bool {
+func (r ResourceRequirement) Matches(name string, gvr schema.GroupVersionResource, resourceLabels map[string]string) bool {
 	// If the requirement does not specify a resource name - only check the
 	// ResourceTypeRequirement i.e. GVR match
+	isMatch := false
```

Comment:
I think you'll find this to be more readable and simpler if we notice that the old code needlessly uses `r.LocalObjectReference.Name` and `r.Name`-- they're the same field.  Also `matches(r.Name, name)` already handles the case of `r.Name == ""`, so the original code is more complex than needed.

Could turn into something like `switch{
case !matches(r.Name, name):
    return false
case !matches(r.ResourceTypeRequiremnt.Matches(gvr):
   return false
}
check the labels...
`
Or could unroll and use matches on r.Group, r.Version, and r.Resource too...

Reply:
Good suggestion. Have taken the simple way out to simplify for now.